mlflow/langchain/retriever_chain.py

```diff
@@ -0,0 +1,139 @@
+"""Chain for wrapping a retriever."""
+from __future__ import annotations
+
+import json
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+from langchain.callbacks.manager import AsyncCallbackManagerForChainRun, CallbackManagerForChainRun
+from langchain.chains.base import Chain
+from langchain.schema import BaseRetriever, Document
+from pydantic import Extra, Field
+from mlflow.utils.annotations import experimental
+
+
+@experimental
+class RetrieverChain(Chain):
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/8808/files#r1275618502, it's probably best to prefix this with `_` so that it doesn't show up in the docs